dev-support/jenkins-scripts/generate-hbase-website.sh

```diff
@@ -125,46 +225,62 @@ FILES_TO_REMOVE=("hbase-*"
                  "images")
 
 for FILE in "${FILES_TO_REMOVE[@]}"; do
-  echo "Removing ${WORKSPACE}/hbase-site/$FILE"
-  rm -Rf "${FILE}"
+  if [ -e "${FILE}" ]; then
+    echo "Removing hbase-site/$FILE"
+    rm -Rf "${FILE}"
+  fi
 done
 
 # Copy in the newly-built artifacts
-cp -au "${WORKSPACE}"/hbase/target/staging/* .
+# TODO what do we do when the site build wants to remove something? Can't rsync because e.g. release-specific docs.
+cp -pPR "${component_dir}"/target/staging/* .
 
 # If the index.html is missing, bail because this is serious
 if [ ! -f index.html ]; then
   echo "The index.html is missing. Aborting."
   exit 1
-else
-  # Add all the changes
-  echo "Adding all the files we know about"
-  git add .
-  # Create the commit message and commit the changes
-  WEBSITE_COMMIT_MSG="Published site at $CURRENT_HBASE_COMMIT."
-  echo "WEBSITE_COMMIT_MSG: $WEBSITE_COMMIT_MSG"
-  git commit -m "${WEBSITE_COMMIT_MSG}" -a
-  # Dump a little report
-  echo "This commit changed these files (excluding Modified files):"
-  git diff --name-status --diff-filter=ADCRTXUB origin/asf-site
-  # Create a patch, which Jenkins can save as an artifact and can be examined for debugging
-  git format-patch --stdout origin/asf-site > "${WORKSPACE}/${CURRENT_HBASE_COMMIT}.patch"
-  echo "Change set saved to patch ${WORKSPACE}/${CURRENT_HBASE_COMMIT}.patch"
-  # Push the real commit
-  git push origin asf-site || (echo "Failed to push to asf-site. Website not updated." && exit -1)
-  # Create an empty commit to work around INFRA-10751
+fi
+
+# Add all the changes
+echo "Adding all the files we know about"
+git add .
+# Create the commit message and commit the changes
+WEBSITE_COMMIT_MSG="Published site at $CURRENT_HBASE_COMMIT."
+echo "WEBSITE_COMMIT_MSG: $WEBSITE_COMMIT_MSG"
+git commit -m "${WEBSITE_COMMIT_MSG}" -a
+# Dump a little report
+echo "This commit changed these files (excluding Modified files):"
+git diff --name-status --diff-filter=ADCRTXUB origin/asf-site | tee "${working_dir}/hbase-file-diff-summary-${CURRENT_HBASE_COMMIT}.txt"
+# Create a patch, which Jenkins can save as an artifact and can be examined for debugging
+git format-patch --stdout origin/asf-site > "${working_dir}/${CURRENT_HBASE_COMMIT}.patch"
+if [ ! -s "${working_dir}/${CURRENT_HBASE_COMMIT}.patch" ]; then
```

Comment:
Is this really the test you want? I think `-s` is testing whether something is a string, but I think you want a file match.

Reply:
`-s` is "file exists and is not zero size"

Reply:
Yeah we determined i was thinking of `-n`.